python/mxnet/optimizer/optimizer.py

```diff
@@ -974,8 +974,7 @@ def update(self, index, weight, grad, state):
         if state is not None:
             mom = state
             mom[:] *= self.momentum
-            grad += wd * weight
-            mom[:] += grad
+            mom[:] += grad + wd * weight
```

Comment:
can you replace with `mom[:] = (self.momentum * mom[:]) + grad + wd * weight` 
and delete line 976. It will be more readable